### PR TITLE
refactor: use single dispatch for DefaultQueryProvider

### PR DESCRIPTION
**Pull request by johnson2427**
### What I did
Added singledispatch to DefaultQueryProvider

### How I did it
Under the config for the BaseInterfaceModel I added singledispatchmethod to keep_untouched
Then added @singledispatchmethod

### How to verify it
Passes original test

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/utils.py [hunk 1]
```diff
@@ -689,7 +689,7 @@ class Config:
 
         #    TypeError: cannot pickle '_thread.RLock' object
 
-        keep_untouched = (cached_property,)
+        keep_untouched = (cached_property, singledispatchmethod)
```
Comment (fubuloubu) [inline src/ape/utils.py:692]: @NotPeopling2day another example of a utility that Pydantic tried to serialize!

File: src/ape/managers/query.py [hunk 2]
```diff
@@ -21,39 +21,25 @@ class DefaultQueryProvider(QueryAPI):
...
     def estimate_query(self, query: QueryType) -> Optional[int]:
-        """
-        Estimates the time that the query will take as a timestamp
-
-        Args:
-            query (``QueryType``): The transaction data you want to query
-
-        Returns:
-             Optional[int]: Depends on whether the query can be completed
-        """
```
Comment (fubuloubu) [inline src/ape/managers/query.py:33]: We removed the docstrings per https://github.com/ApeWorX/ape/pull/535#discussion_r820031929

@unparalleled-js do you have any thoughts here in terms of docs rendering?

Comment (fubuloubu) [reply src/ape/managers/query.py:33]: Also note that this class will eventually move to a 1st party plugin

**Review by fubuloubu — APPROVED** [review]

**Comment by johnson2427** [comment]
Good to merge?

**Comment by fubuloubu** [comment]
> Good to merge?

Wait for @unparalleled-js to respond to the docs question above, if that looks good then good to merge

File: src/ape/managers/query.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/managers/query.py:33]: Yes, these query subclasses **do not** need documentation unless they wildly differ from the API

**Review by unparalleled-js — APPROVED** [review]
LGTM

**Comment by johnson2427** [comment]
Sorry guys, I misread the answer from @unparalleled-js

**Review by unparalleled-js — APPROVED** [review]
didn't mean to re-request fubuloubu!

**Pull request merged**